From 0000000000000000000000000000000000000001 Mon Sep 17 00:00:00 2001
From: Dusk Player <dusk@localhost>
Date: Fri, 3 Jul 2026 09:00:00 +0200
Subject: [PATCH 14/14] src/posix: do not use dup3/pipe2 on Apple platforms

Apple OSes do not provide dup3() or pipe2(), but newer iOS SDKs
(observed with the iOS 26.5 simulator SDK / Xcode 26.6) let the
configure link checks for them succeed, so HAVE_DUP3/HAVE_PIPE2 get
defined and the build fails with implicit-function-declaration errors
in src/posix/filesystem.c. Guard both call sites so the portable
fallbacks are used on Apple platforms regardless of what configure
detected.
---
 src/posix/filesystem.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/posix/filesystem.c b/src/posix/filesystem.c
--- a/src/posix/filesystem.c
+++ b/src/posix/filesystem.c
@@ -190,7 +190,7 @@
 
 int vlc_dup2(int oldfd, int newfd)
 {
-#ifdef HAVE_DUP3
+#if defined(HAVE_DUP3) && !defined(__APPLE__)
     return dup3(oldfd, newfd, O_CLOEXEC);
 #else
     int ret = dup2(oldfd, newfd);
@@ -202,7 +202,7 @@
 
 int vlc_pipe (int fds[2])
 {
-#ifdef HAVE_PIPE2
+#if defined(HAVE_PIPE2) && !defined(__APPLE__)
     return pipe2(fds, O_CLOEXEC);
 #else
     int ret = pipe(fds);
--
2.39.5 (Apple Git-154)
